modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequenceFactory.java

```diff
@@ -78,4 +78,21 @@ public CoordinateSequence create(int size, int dimension) {
       return new CoordinateArraySequence(size);
     return new CoordinateArraySequence(size, dimension);
   }
+  
+  public CoordinateSequence create(int size, int dimension, int measures) {
+    if (dimension > 4) {
+      dimension = 4;
+      //throw new IllegalArgumentException("dimension must be <= 4");
+    }
+    if (measures > 1) {
+      measures = 1;
+      //throw new IllegalArgumentException("measures must be <= 1");
+    }
+    if (dimension < 2)
+      dimension = 2; // handle bogus dimension
+    if (dimension - measures < 2) {
+      throw new IllegalArgumentException("max spatial dimension 2 required");
```

Comment:
Do you mean 'Minimum spatial dimension of 2 required'?  (I'm also a fan of dumping the inputs.  E.g. "Minimum spatial dimension of 2 required.  Input dimension was $dimension, input number of measures was $measures.")

Reply:
fixed